app/src/main/java/edu/uc/reedws/musiclink/MainActivity.kt

```diff
@@ -56,18 +60,31 @@ class MainActivity() : AppCompatActivity() {
 
     // Method to show Dialog Screen for adding a new playlist
     private fun showAlertDialog(viewModel: ApplicationViewModel) {
-
         val addPlaylistDialogBuilder = AlertDialog.Builder(this)
         val inflater = layoutInflater
         addPlaylistDialogBuilder.setTitle("Enter name of playlist to create")
         val dialogLayout = inflater.inflate(R.layout.add_playlist_dialog, null)
-        val editText  = dialogLayout.findViewById<EditText>(R.id.editText)
+        val searchedSong  = dialogLayout.findViewById<EditText>(R.id.searchedSong)
+
         addPlaylistDialogBuilder.setView(dialogLayout)
-        addPlaylistDialogBuilder.setPositiveButton("Done") { dialogInterface, i -> Toast.makeText(applicationContext, "You added " + editText.text.toString(), Toast.LENGTH_SHORT).show()
-            val newPlaylistName = editText.text
+
+        addPlaylistDialogBuilder.setPositiveButton("Done") { dialogInterface, i ->
+            if(searchedSong.length() > 0) {
+            Toast.makeText(
+                applicationContext,
+                "You added " + searchedSong.text.toString(),
+                Toast.LENGTH_SHORT
+            ).show()
+
+            val newPlaylistName = searchedSong.text
             viewModel.createPlaylist(newPlaylistName.toString())
+            }
+            else {
+                Toast.makeText(applicationContext, "You have not entered anything", Toast.LENGTH_SHORT).show()
+            }
         }
-        addPlaylistDialogBuilder.setNeutralButton("Cancel") { dialog, id -> dialog.cancel()}
+        addPlaylistDialogBuilder.setNeutralButton("Cancel") { dialog, id -> dialog.cancel() }
+
```

Comment:
I ran an application and I entered nothing inside the add playlist text field, but the application still added a playlist with no name instead of prompting the user that they have not entered anything. So I added a empty string validation check to make sure that the user enters something in the searched song field. If they do enter something, perform the regular operation, but if they don't, then create a toast that tells the user that they have not entered anything and do not create a playlist.